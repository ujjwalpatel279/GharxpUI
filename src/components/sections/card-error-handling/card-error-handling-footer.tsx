import React, { FunctionComponent, ReactElement } from 'react';
import { useHistory } from 'react-router-dom';

import { CardFooter } from '../../shared/card/card-footer';
import { Button } from '../../shared/button/button';

import './card-error-handling.scss';
interface CardErrorHandlingFooterInterface {
  url?: string;
}
export const CardErrorHandlingFooter: FunctionComponent<CardErrorHandlingFooterInterface> = ({ url }): ReactElement => {
  const history = useHistory();

  const tryAgain = () => {
    if (url && url.length > 1) {
      history.push(url);
    } else {
      history.goBack;
    }
  };

  return (
    <CardFooter>
      <footer>
        <div>
          <Button type="button" variant="primary" childClasses="full-width" onClick={tryAgain}>
            Try Again
          </Button>
        </div>
      </footer>
    </CardFooter>
  );
};
