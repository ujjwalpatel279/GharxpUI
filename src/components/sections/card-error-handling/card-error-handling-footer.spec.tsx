import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { CardErrorHandlingFooter } from './card-error-handling-footer';

describe('shows card bad redirects footer', () => {
  afterEach(cleanup);
  it('show bad redirects card footer', () => {
    const { asFragment } = render(<CardErrorHandlingFooter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
