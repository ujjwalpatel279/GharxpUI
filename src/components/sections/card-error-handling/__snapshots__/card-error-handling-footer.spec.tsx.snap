// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows card bad redirects footer show bad redirects card footer 1`] = `
<DocumentFragment>
  <div
    class="c-card__footer "
  >
    <footer>
      <div>
        <button
          class="c-btn c-btn--primary full-width"
          type="button"
        >
          <span>
            Try Again
          </span>
        </button>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
